include/stout/flags/flags.cc

```diff
@@ -663,6 +683,62 @@ void Parser::Parse(const std::vector<ArgumentInfo>& values) {
 
 ////////////////////////////////////////////////////////////////////////
 
+void Parser::SetFieldMessageOrAggregateErrors(
+    const std::string& value,
+    const std::string& name,
+    const google::protobuf::FieldDescriptor* field,
+    google::protobuf::Message* message,
+    std::set<std::string>& errors) {
+  // Parse the value using an error collector that aggregates the
+  // error for us to print out later.
+  struct ErrorCollector : public google::protobuf::io::ErrorCollector {
+    void AddError(
+        int /* line */,
+        int /* column */,
+        const std::string& message) override {
+      error += message;
+    }
+
+    void AddWarning(
+        int line,
+        int column,
+        const std::string& message) override {
+      // For now we treat all warnings as errors.
+      AddError(line, column, message);
+    }
+
+    std::string error;
+  } error_collector;
+
+  google::protobuf::TextFormat::Parser text_format_parser;
+  text_format_parser.RecordErrorsTo(&error_collector);
+  if (!text_format_parser.ParseFieldValueFromString(
+          value,
+          field,
+          message)) {
+    errors.insert(
+        "Failed to parse flag '" + name
+        + "' from normalized value '" + value
+        + "' due to protobuf text-format parser error(s): "
+        + error_collector.error);
+  } else {
+    // Successfully parsed!
+    parsed_[field] = {name, value};
+  }
+}
+
+////////////////////////////////////////////////////////////////////////
+
+std::string Parser::GetNormalizedDefaultValue(
+    const std::string& value,
+    bool has_type_string) {
+  return (!has_type_string)
```

Comment:
I'd recommend doing "positive boolean" checks, where you reverse this so you don't need to negate the boolean:

so prefer:
```
return foo ? b : a;
```

over:
```
return !foo ? a : b;
```

Similarly, maybe prefer comparisons with `==` over those with `!=`